### PR TITLE
[CPU] add checking on attention_mask of SDPA

### PR DESCRIPTION
**Pull request by tiger100256-hu**
### Details:
 - *also check the attention mask in the shape infer of ScaledDotProductAttentionWithKVCache*

### Tickets:
 - *CVS-161604*

### CODE REVIEW

**Comment by maxnick** [comment]
@luo-cheng2021 , could you please review?

File: src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp [hunk 1]
```diff
@@ -52,6 +76,32 @@ class SDPAShapeInfer : public ShapeInferEmptyPads {
             output_dims[i] = query_dims[permute_axes[i]];
         }
 
+        if (input_shapes.size() == 7 && !m_config.is_causal) {
```
Comment (luo-cheng2021) [inline src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp:79]: May we add a lambda function and call it twice to reuse the similar logic?

File: src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp [hunk 2]
```diff
@@ -22,11 +22,35 @@ class SDPAShapeInfer : public ShapeInferEmptyPads {
         VectorDims present_v_dims = input_shapes.back().get();
         const auto& beam_idx_dims = input_shapes.end()[-3].get();
         const auto& permute_axes = m_config.permute_axes;
-
         if (permute_axes.empty()) {
             // [B, H, L, S]
             present_v_dims[0] = beam_idx_dims[0];
             present_v_dims[2] += query_dims[2];
+            if (input_shapes.size() == 7 && !m_config.is_causal) {
+                const auto& attn_mask_dims = input_shapes[3].get();
```
Comment (luo-cheng2021) [inline src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp:30]: From the [spec](https://docs.openvino.ai/2025/documentation/openvino-ir-format/operation-sets/operation-specs/sequence/scaled-dot-product-attention.html), the dimension of attention mask may be 2+.

File: src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp (hunk 1 above)
Comment (tiger100256-hu) [reply src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp:37]: fix, when permute_axes.empty() is empty, I make permute_axes = {0 ,1 ,2, 3}, so it make the code simpler

File: src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp (hunk 2 above)
Comment (tiger100256-hu) [reply src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp]: fix, and add more test case

**Review by luo-cheng2021 — APPROVED** [review]
LGTM. Thanks.

**Comment by yuxu42** [comment]
Hi @v-Golubev @maxnick could you please take a review? Thanks!

File: src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp [hunk 3]
```diff
@@ -18,31 +18,13 @@ class SDPAShapeInfer : public ShapeInferEmptyPads {
...
-            auto output_dims = query_dims;
-            output_dims[3] = present_v_dims[3];
-            auto present_k_dims = present_v_dims;
-            present_k_dims[3] = query_dims[3];
-            return {{output_dims, present_k_dims, present_v_dims}, ShapeInferStatus::success};
-        }
-
+        const auto& beam_idx_dims = input_shapes[inputs_size - 3].get();
+        const auto& permute_axes_origin = m_config.permute_axes;
+        auto permute_axes = (permute_axes_origin.empty()) ? std::vector<size_t>{0, 1, 2, 3} : permute_axes_origin;
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp:27]:
```suggestion
        static const std::vector<size_t> direct_axis_order = {0, 1, 2, 3};
        const auto& permute_axes = (permute_axes_origin.empty()) ? direct_axis_order : permute_axes_origin;
```
This allows to avoid a copy during instancing of the `permute_axes` object, which is in fact read only.
Also this allows to avoid a malloc, each time `permute_axes_origin.empty()` is `true`.

File: src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp [hunk 4]
```diff
@@ -18,31 +18,13 @@ class SDPAShapeInfer : public ShapeInferEmptyPads {
 
     IShapeInfer::Result infer(const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
                               [[maybe_unused]] const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
-        OPENVINO_ASSERT(input_shapes.size() >= 3,
-                        "SDPAShapeInfer: expected at least 3 inputs, got ",
-                        input_shapes.size());
+        auto inputs_size = input_shapes.size();
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp:21]: Better to declare as `const`

File: src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp (hunk 3 above)
Comment (tiger100256-hu) [reply src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp]: fixed

File: src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp (hunk 4 above)
Comment (tiger100256-hu) [reply src/plugins/intel_cpu/src/shape_inference/custom/scaled_attn.cpp]: fixed

**Review by maxnick — APPROVED** [review]
